velox/common/testutil/CMakeLists.txt

```diff
@@ -12,8 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-velox_add_library(velox_test_util ScopedTestTime.cpp TestValue.cpp)
-velox_link_libraries(velox_test_util PUBLIC velox_exception)
+velox_add_library(velox_test_util ScopedTestTime.cpp TestValue.cpp OutputMatcher.cpp)
+velox_link_libraries(velox_test_util PUBLIC velox_exception Folly::folly GTest::gtest re2::re2)
```

Comment:
`velox_test_util` is used in several non-test targets, most prominently `velox_exec`, linking it to GTest will break builds with `VELOX_BUILD_TESTING=OFF`.

The main issue is probably the use of a test util in non-test targets but I don't know what it's used for and how involved a refactor would be. cc @pedroerp

Reply:
Would it be better to put it in velox/common/testutil/tests then?

Reply:
Btw the name `velox/common/testutil/` is confusing. It seems to suggest the code under it would be used in tests shared by all modules.

Reply:
I agree that the name is confusing, it should either only be used in tests (then it can link against gtest just fine) or it should get a new name to avoid confusion... but as I said, I can't really be of help refactoring this, hence my ping to Pedro.

Reply:
Seems like this one is the TestValue library that we need to link to prod libraries to allow gtest to inject hooks. It should not have any dependency on gtest and other test code though, so I think we could rename it to `velox_test_value` to avoid confusion?

We could even go as far as renaming the TestValue class name to something without the "test" on the name ("ValueHook" or similar). Cc: @xiaoxmeng who has more context on this.

Maybe create an Issue so we can follow up there on next steps?

Reply:
@assignUser @pedroerp Thanks for your inputs. I will separate TestValue.cpp out tomorrow and create an issue.

Reply:
@pedroerp we might rename to VELOX_TEST_VALUE as it is used for testing and have TEST in name to make it clear.

Reply:
@xiaoxmeng I'm not sure what you meant because if it's named VELOX_TEST_VALUE then people may still think it could be linked to gtest.
So I went one step back: I restored the original velox_test_util target, and added a new target velox_test_output_matcher that only has OutputMatcher.cpp:
```
velox_add_library(velox_test_util ScopedTestTime.cpp TestValue.cpp)
velox_link_libraries(velox_test_util PUBLIC velox_exception)

velox_add_library(velox_test_output_matcher OutputMatcher.cpp)
velox_link_libraries(velox_test_output_matcher PUBLIC Folly::folly GTest::gtest re2::re2)
```
Do you think this works?